src/sensors/linearServo/LinearServo.java

```diff
@@ -31,19 +31,42 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
         currentPos = this.maxLength * this.get();
     }
 
+    /** calibrate the linear servo
+     * you can override this function and use different position or use setPosition once.
+     *
+     * the calibrate position doesn't meter and you only need to do this once*/
+    public void calibrate(){
+        setPosition(0);
+    }
+
     /**
      * @param setpoint the target position of the servo [mm]
      */
     public void setPosition(double setpoint) {
         setPos = MathUtil.clamp(setpoint, 0, maxLength);
-        setSpeed((setPos / maxLength * 2) - 1);
+        super.setSpeed((setPos / maxLength * 2) - 1);
+    }
+
+    /** move the linear servo forward or reverse according to the speed sign
+     *
+     *@param speed determine the direction
+     *
+     * speed equal to 0 stops the linear servo
+     * speed > 0 moves the linear servo forward
+     * speed < 0 moves the linear servo reverse
+     *
+     * dont forget to calibrate once the linearServo other wise it will not stop.
+     * */
+    public void setSpeed(double speed){
+        setPosition(speed == 0 ? currentPos : speed > 0 ? maxLength : 0);
```

Comment:
I still dont agree about this way, but if you dont fix it next commit ill approve it..